bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/CommunicationManager.java

```diff
@@ -467,6 +469,9 @@ && channelTypeDefinesDimension(channel.getChannelTypeUID())) {
 
         if (acceptedTypes.contains(originalType.getClass())) {
             return originalType;
+        } else if (acceptedTypes.contains(OnOffType.class) && originalType instanceof State state
+                && PercentType.class.isAssignableFrom(originalType.getClass())) {
+            return (@Nullable T) state.as(OnOffType.class);
         } else {
```

Comment:
Should we also add code here for `PercentType.class` so that we can link a `Color` item to a `Dimmer` channel in the same way?

Reply:
Ok, I think I now understood what you meant - and it should fix the tests.